cirq-core/cirq/value/measurement_key.py

```diff
@@ -112,6 +139,13 @@ def with_key_path_prefix(self, path_component: str):
         return self._with_key_path_((path_component,) + self.path)
 
     def _with_measurement_key_mapping_(self, key_map: Dict[str, str]):
-        if self.name not in key_map:
+        base_key = self.name if self.name else default_measurement_key(self.qubits)
+        if base_key not in key_map:
             return self
-        return self.replace(name=key_map[self.name])
+        return self.replace(name=key_map[base_key])
+
+    def with_qubits(self, qubits: Tuple['cirq.Qid', ...]):
+        """Updates the MeasurementKey to operate on the input qubits.
+
+        Functionally no-op if the MeasurementKey already has a name."""
```

Comment:
Actually removed this comment from here since there is a risk of creating more confusion than clarity (for e.g. regarding the comment above). It is not really a "no-op" since the object is changed, it's just functionally same in the current state. Added more clarification in the `qubits` arg comment for the class.